### DISCUSSION TITLE
Using docker-compose file for "Apache (prod)" leads to "Allowed memory size exhausted"

### DISCUSSION THREAD

**Discussion by meinrecht** [kimai/kimai#5078] [Docker] [answered]
### Describe the problem

After successfully using the image `kimai/kimai2:apache-dev` I wanted to switch to `kimai/kimai2:apache` as per the official documentation here https://www.kimai.org/documentation/docker-compose.html
My `docker-compose.yml` differs only in regard to passwords, the external port mapping and the "TRUSTED_HOSTS".
Besides the line containing "version" is omitted, as I use `docker compose` instead of `docker-compose`.

After the database but before `apache` is started I get the following error:

> // Warming up the cache for the prod environment with debug false
kimai-1  |
kimai-1  | PHP Fatal error:  Allowed memory size of 268435456 bytes exhausted (tried to allocate 135168 bytes) in /opt/kimai/vendor/symfony/routing/Loader/AttributeFileLoader.php on line 82
kimai-1  | [2024-09-04T16:00:18.088254+00:00] deprecation.INFO: User Deprecated: Since twig/twig 3.12: Twig Filter "spaceless" is deprecated in /opt/kimai/vendor/kevinpapst/tabler-bundle/templates/components/actions.html.twig at line 19. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Since twig/twig 3.12: Twig Filter \"spaceless\" is deprecated in /opt/kimai/vendor/kevinpapst/tabler-bundle/templates/components/actions.html.twig at line 19. at /opt/kimai/vendor/twig/twig/src/ExpressionParser.php:832)"} {"channel":"deprecation"}
kimai-1  | [2024-09-04T16:00:18.145198+00:00] deprecation.INFO: User Deprecated: Since twig/twig 3.12: Twig Filter "spaceless" is deprecated in /opt/kimai/vendor/kevinpapst/tabler-bundle/templates/components/button.html.twig at line 4. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Since twig/twig 3.12: Twig Filter \"spaceless\" is deprecated in /opt/kimai/vendor/kevinpapst/tabler-bundle/templates/components/button.html.twig at line 4. at /opt/kimai/vendor/twig/twig/src/ExpressionParser.php:832)"} {"channel":"deprecation"}
kimai-1  | [2024-09-04T16:00:18.723785+00:00] php.CRITICAL: Fatal Error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 135168 bytes) {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\OutOfMemoryError(code: 0): Error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 135168 bytes) at /opt/kimai/vendor/symfony/routing/Loader/AttributeFileLoader.php:82)"} {"channel":"php"}
kimai-1  | PHP Fatal error:  Allowed memory size of 268435456 bytes exhausted (tried to allocate 32768 bytes) in /opt/kimai/vendor/symfony/var-dumper/Cloner/AbstractCloner.php on line 202
kimai-1  | + chown -R 33:33 /opt/kimai/var
kimai-1  | + '[' -e /use_apache ']'
kimai-1  | + exec /usr/sbin/apache2 -D FOREGROUND
kimai-1  | [Wed Sep 04 17:00:18.832025 2024] [mpm_prefork:notice] [pid 1:tid 1] AH00163: Apache/2.4.61 (Debian) configured -- resuming normal operations

There is very little inside the database, as we just made some test entries.

Besides the error message itself I did not notice any difficulties, but I suppose that depends on what I do.

The error does not occur with `apache-dev`.

### Describe your setup and add your Docker compose file (redact your credentials)

OS: Linux, Debian 12.6
Docker version 27.1.1
docker compose version v2.29.1

Docker compose file as per the documentation for "Apache (prod)" minus the changes mentioned above.

### Command used to run the container

docker compose up -d

**Comment by kevinpapst** [comment, marked as answer]
Have you tried setting the environment variable `memory_limit` to something like `512M`?

See https://www.kimai.org/documentation/docker.html#environment-variables--runtime-args

**Comment by meinrecht** [comment]
Thank you, that worked. 
Sorry, for bothering you. I somehow missed this setting.
